torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -1360,13 +1380,10 @@ size_t TensorPipeAgent::timeoutMapSize() {
 }
 
 size_t TensorPipeAgent::numPendingResponses() {
-  size_t totalPending = 0;
-  std::unique_lock<std::mutex> lock(mutex_);
-  for (const auto& entry : connectedPipes_) {
-    totalPending += entry.second.pendingResponseMessage_.size();
-  }
-  return totalPending;
```

Comment:
@pritamdamania87 You added this function in #50564, could you confirm the version I'm proposing is equivalent? I prefer this new version because the old one would now require acquiring multiple mutexes (all the ones of all the ClientPipe objects), which would make them more contended and risk deadlocks. Thanks!

Reply:
Yes, I think this is equivalent.